### PR TITLE
ispell: fix hash and drop patches

### PR DESCRIPTION
**Pull request by tadeokondrak**
###### Motivation for this change

https://github.com/NixOS/nixpkgs/pull/64285#issuecomment-580987532

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lheckemann** [comment]
If we're not applying the patches anymore, we should probably also remove them from nixpkgs.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
See @lheckemann comment

File: pkgs/tools/text/ispell/default.nix [hunk 1]
```diff
@@ -6,25 +6,11 @@ stdenv.mkDerivation rec {
...
-    ./patches/0008-Tex-backslash.patch
-    ./patches/0009-Fix-FTBFS-on-glibc.patch
-    ./patches/0011-Missing-prototypes.patch
-    ./patches/0012-Fix-getline.patch
-    ./patches/0013-Fix-man-pages.patch
-    ./patches/0021-Fix-gcc-warnings.patch
-    ./patches/0023-Exclusive-options.patch
-    ./patches/0024-Check-tempdir-creation.patch
-    ./patches/0025-Languages.patch
-    ./patches/0030-Display-whole-multibyte-character.patch
```
Comment (jtojnar) [inline pkgs/tools/text/ispell/default.nix]: Some of these still might make sense when updated from https://salsa.debian.org/debian/ispell/tree/debian/3.4.00-8/debian/patches. Why they did not upstream them even after all these years is beyond me.

Or if this is intended only for the German dictionary we might make it local as [originally suggested](https://github.com/NixOS/nixpkgs/pull/36155#issuecomment-369531974).

Or maybe we are fine with vanilla ispell, even though some of those patches look like they should definitely be upstreamed.

**Comment by Ekleog** [comment]
I'd vote for just landing this, up-to-date vanilla ispell probably is better matching the nixos way of doing things than debian-old ispell.

@jtojnar What do you think about merging this as is?

**Comment by jtojnar** [comment]
I would definitely prefer vanilla package but would like still

@GrahamcOfBorg build hunspellDicts.de-de hunspellDicts.de-ch hunspellDicts.de-at

**Comment by timor** [comment]
I cannot comment on these patches, sorry. I merely re-used @bennofs' code (for the sole purpose of building the German hunspell dictionaries).

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 79025` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>hunspellDicts.de-at (hunspellDicts.de_AT)</li>
    <li>hunspellDicts.de-ch (hunspellDicts.de_CH)</li>
    <li>hunspellDicts.de-de (hunspellDicts.de_DE)</li>
    <li>ispell</li>
  </ul>
</details>

**Pull request merged**